neard/src/runtime.rs

```diff
@@ -1370,9 +1370,15 @@ impl RuntimeAdapter for NightshadeRuntime {
             && chunk_epoch_id != head_next_epoch_id)
     }
 
+    #[cfg(feature = "protocol_feature_evm")]
     fn evm_chain_id(&self) -> u128 {
         self.genesis_config.evm_chain_id
     }
+
+    #[cfg(not(feature = "protocol_feature_evm"))]
+    fn evm_chain_id(&self) -> u128 {
```

Comment:
Do we want to leak this part? What prevent us from only including it in the trait only if the feature is enabled? @evgenykuzyakov

Reply:
This is a minimum change to make evm_chain_id a noop when without protocol_feature_evm. Alternative is find all ApplyState and guard with feature there (1 definition, 7 constructings)